src/main/java/com/mraof/minestuck/client/gui/playerStats/EcheladderScreen.java

```diff
@@ -37,105 +37,80 @@ public class EcheladderScreen extends PlayerStatsScreen
-	private static final int rows = 12;
+	private static final int LADDER_X_OFFSET = 163, ladderYOffset = 25;
+	private static final int ROWS = 12;
+	private static final int RUNG_Y = 14;
 	
-	private static final int timeBeforeAnimation = 10, timeBeforeNext = 16, timeForRung = 4, timeForShowOnly = 65;
+	private static final int GREY = 0x404040;
+	private static final int BLUE = 0x0094FF;
 	
+	private static final int TIME_BEFORE_ANIMATION = 10, TIME_BEFORE_NEXT = 16, TIME_FOR_RUNG = 4, TIME_FOR_SHOW_ONLY = 65;
+	private static final int TIME_TILL_NEXT = TIME_BEFORE_NEXT + TIME_FOR_RUNG;
+	
+	private final int maxScroll;
 	private int scrollIndex;
 	private boolean isScrolling;
 	
-	public static int lastRung = -1;	//The current rung last time the gui was opened. Used to determine which rung to display increments from next time the gui is opened
-	public static int animatedRung;	//The rung animated to or the latest to be animated
-	private int fromRung;	//First rung to display increments from; (actually the one right before that one)
-	private int animationCycle;	//Ticks left on the animation cycle
-	private int animatedRungs;	//The amount of rungs to animate
+	public static int lastRung = -1;    //The current rung last time the gui was opened. Used to determine which rung to display increments from next time the gui is opened
+	public static int animatedRung;    //The rung animated to or the latest to be animated
+	private final int finalRung;
+	private boolean showLastRung = true;
+	private int fromRung;    //First rung to display increments from; (actually the one right before that one)
+	private int animationCycle;    //Ticks left on the animation cycle
+	private int animatedRungs;    //The amount of rungs to animate
 	
 	public EcheladderScreen()
 	{
 		super(Component.translatable(TITLE));
 		guiWidth = 250;
 		guiHeight = 202;
+		
+		finalRung = Rungs.finalRung();
+		
+		maxScroll = (finalRung + 1) * RUNG_Y - 154;
 	}
 	
 	@Override
 	public void init()
 	{
 		super.init();
-		scrollIndex = Mth.clamp((ClientPlayerData.getRung() - 8)*14, 0, MAX_SCROLL);
-		animatedRung = Math.max(animatedRung, lastRung);	//If you gain a rung while the gui is open, the animated rung might get higher than the lastRung. Otherwise they're always the same value.
+		scrollIndex = Mth.clamp((ClientPlayerData.getRung() - 8) * RUNG_Y, 0, maxScroll);
+		animatedRung = Math.max(animatedRung, lastRung);    //If you gain a rung while the gui is open, the animated rung might get higher than the lastRung. Otherwise they're always the same value.
 		fromRung = lastRung;
 		lastRung = ClientPlayerData.getRung();
 	}
 	
 	@Override
 	public void render(GuiGraphics guiGraphics, int mouseX, int mouseY, float partialTicks)
 	{
-		updateScrollAndAnimation(mouseX, mouseY);
+		updateScrollAndAnimation(mouseY);
 		
 		int speedFactor = MinestuckConfig.CLIENT.echeladderAnimation.get().getSpeed();
 		int currentRung;
-		boolean showLastRung = true;
```

Comment:
This was changed to a field, but in the process does not get properly reset. Before, it would be true unless explicitly reassigned by the code below. Now, it is whatever it was last tick (which may be false) unless explicitly reassigned in getCurrentRung(). This might be your rung animation issue.

Reply:
Fixed.